src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -631,13 +632,16 @@ static unsigned int icap_get_clock_frequency_counter_khz(const struct icap *icap
 			while (times != 0) {
 				status =
 				    reg_rd(icap->icap_clock_freq_counters[idx]);
-				if (status == 0x2)
+				if ((status & 0xffff) == 0x2)
 					break;
 				mdelay(1);
 				times--;
 			};
-			freq = reg_rd(icap->icap_clock_freq_counters[idx] +
-				OCL_CLK_FREQ_COUNTER_OFFSET);
+			if ((status & 0xffff) == 0x2) {
+				freq = (status & 0x10000) ?
```

Comment:
Could you define some meaningful macros for bits in status registers for "0x2" and "0x10000"?

Reply:
done